src/amity.py

```diff
@@ -0,0 +1,623 @@
+		for person in self.people:
+			if person.uid == uid:
+				return person
+		return False
+
+
+	def remove_person_from_room(self, person,room_type):
+		"""remove person from a room"""
+
+		room_name = person.assigned_room[room_type]
+		room = self.get_room_by_name(room_name)
+		if room:
+			if room[0].remove_person(person):
+				self.exception_room = room[0]
+				return room[0]
+
+		return False
+
+	def load_people(self, args):
+		"""creates people from a specified file and allocates each person based on person type"""
+
+		path = args['<file_location>']
+		if not Util.is_file(path):
+			Util.print_line('File location is invalid')
+			return False
+
+		with open(path,'r') as file:
+			for line in file:
+				records = line.split()
+				living_space = False
+				firstname = records[0]
+				lastname = records[1]
+				person_type = records[2]
+				if person_type.upper() == 'FELLOW':
+					if(records[3].upper() == 'Y'):
+						living_space = True
+					person = Fellow(firstname,lastname,living_space)
+				else:
+					person = Staff(firstname,lastname)
+
+				self.people.append(person)
+			  	Util.print_line(person.name() + ' succesfully created')
+			  	self.allocate(person,None,'OFFICE')
+	  			if person.is_fellow():
+	  				self.allocate(person)
+		  	self.save_state_to_pickle()
+
+	def get_db_name(self, args):
+		"""return specified db name from arguments if available.
+		return amity if --db options is not specified"""
+
+		db_name = 'amity'
+		if args['--db']:
+			db_name = args['--db']
+
+		return db_name
+
+	def save_state(self, args):
+		"""installs necessary tables if not created.
+		saves current state to specified sqlite db name"""
+
+		db_name = self.get_db_name(args)
+		migrate = Migration(db_name)
+		if self.settings['drop_db']:
+			migrate.drop()
+			self.settings['drop_db'] = False
+			self.set_file_location('config.pkl')
+			self.pickle_dump(self.settings)
+
+		migrate.install()
+		self.save_room_state(db_name)
+		self.save_people_state(db_name)
+
+	def save_room_state(self, db_name='amity'):
+		"""saves current rooms state to sqlite and deletes rooms pickle file"""
+
+		if len(self.rooms) == 0:
+			Util.print_line('No room to save')
+			return False
+
+		for room in self.rooms:
+			room.set_db(db_name)
+			if room.save():
+				Util.print_line(room.name + ' save!')
+			else:
+				Util.print_line(room.name + ' already exists')
+
+		self.set_file_location('rooms.pkl')
+		self.remove()
+
+	def load_room_state(self, db_name='amity'):
+		"""loads room state from sqlite db"""
+
+		db = Db(db_name,'room')
+		rooms = db.find_all()
+		if not rooms:
+			Util.print_line('No rooms record found')
+			return False
+			
+		for row in rooms:
+			if row['type'] == 'OFFICE':
+				room = Office(str(row['name']))
+			else:
+				room = LivingSpace(str(row['name']))
+
+			if int(row['allocated']) > 0:
+				db.table_name = 'person'
+				people = db.execute('SELECT * FROM person WHERE assigned_room LIKE "%'+ room.name +'%"')
```

Comment:
**Code Quality:** use of `.format()` is a cleaner way of concatenating strings that using `+`.